<launch>



  <!-- Launch Hector SLAM for real-time mapping -->
  <include file="$(find navigation)/launch/hector.launch" />

  <!-- Launch Move Base (global & local planner) -->
  <include file="$(find navigation)/launch/move_base.launch" />

  <!-- Launch Heading Error Node (based on local plan and current pose) -->
  <node pkg="navigation" type="local_plan_heading_node.py" name="local_plan_heading_node" output="screen">
    <param name="publish_topic" value="/message/ArduinoROS" />
    <param name="goal_tolerance" value="0.1" />
  </node>
    <!-- Launch Communication node to Arduino-->
  <node pkg="lane_pkg" type="send_serial_node.py" name="send_serial_node" output="screen" />
  
  <!-- Optional: Visualize in RViz--> 
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigation)/rviz/navigation.rviz" required="false" />

</launch>
